config.txt
<?php
// CONFIG.PHP - ListMail Configuration File
//
// This is a .PHP file that must maintain proper syntax.  Be careful to only
// change the values between the single quotes and nothing else, or the script
// may not work correctly.
//
// Required is your database information, the host should normally be left at the
// setting, 'localhost', but not always.  Check with your host if it doesn't work.

// Enter your MySQL Information!
$sqlhost = 'localhost';
$sqluser = 'kh86';
$sqlpass = 'envymask';
$sqldb = 'laikaope_ListMailPro';

// SAVE this file as config.php, -not- config.php.txt
// UPLOAD all of the files to a new folder on your web site, such as public_html/mail/
// BROWSE to the new subdirectory

// OPTIONS are available for different server types
// These options are rarely needed but are available if problems occur
// To change options remove and add the comment characters "//" from the beginning of the lines.
// Examples are provided with their defaults
// Setting an option to '' means no value, usually off.  Be careful to keep proper PHP syntax.

// Detect line endings for Mac browsers, default 0
// ini_set('auto_detect_line_endings',1);

// Use -f setting to set return-path correctly with PHP mail()
// This is used for welcome and confirmation messages.  If you are receiving "raw" bounce
// messages from welcome emails even if you have bounce configured this may need to be set.
$phpmailf = '';
// $phpmailf = 1;

// International character set, default iso-8859-1 if not set)
$charset = '';
// $charset = 'iso-8859-1';

// Manually set path to MySQL binaries - for problems with backup / restore.
// The linux commandline utilities "mysql" and "mysqldump" need to be available for the
// backup process.  If ListMail cannot detect the path you can set it manually.

// $sqlpath = "/usr/bin/mysql/";
// $sqlpath = "/usr/local/bin/mysql/";
// $sqlpath = "/usr/local/mysql/bin/";
// $sqlpath = "/usr/local/psa/mysql/bin/";

// max random value for problematic servers (deprecated)
// $randmaxx = '50000';

// mail counter increment, default 50
// $mailcount = 50;

// path for mimeclass/mimepart (sendwelcome)
// blank is normal - use current directory always '.' comment to allow setting $mp from your script
$mp = '';

// add Maximizer option to exports
/* $maximizer = 1;
$max_header = '"MTI: Web Inquiry"
"BUSINESS_PEOPLE","1","401","E-mail Address"
"P_E-mail Address","FIRST NAME","LAST NAME","ADDRESS LINE 1","ADDRESS LINE 2","CITY","COUNTY","POSTCODE","COUNTRY","PHONE 1"
"ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-59","ALPHA-21","ALPHA-21","ALPHA-21","ALPHA-119"';
$max_fields = 'em,fn,ln,u1,u2,u3,u4,u5,u6,u7'; */

// !! Do not change anything below here !!
if (get_magic_quotes_gpc()==1) { $gpc = 'yes'; }
?>


Changes.txt
Changes v1.87d -> v1.88

 - fixed a bug with the installation login cookie
 - fixed global message codes supserseding list-specific codes 
 - improved email validation function - see http://www.linuxjournal.com/article/9585
 - added an alternative version of bounce.cgi that uses POST and does not require a system utility

note: you can replace just signup.php, editconf.php and admin.php.  all changes will take effect without any notification as there is no database update required

-----------------------------------------------------------------

Changes v1.87 -> v1.87d

 - fixed 'Update all users on this followup to new delay' feature
 - fixed 'Always write SMTP log' feature - was not writing log during dailymail
 - fixed display issue with 'skip bounce via mailbox' checkbox on dailymail page
 - fixed display issue with 'you have items in outgoing queue'
 - added list 'additional options' processing when importing users
 + fixed a bug in this feature's implementation
 - fixed 'move users who reach end of followup sequence' list followup option
 - fixed error when scheduling email to more than one user selection on the same day
 - improved security of upgrade.php script

-----------------------------------------------------------------

Changes v1.86b -> v1.87

 - added followup options for moving and removing users who reach end of sequence
 - added meta 'robots' tag with 'noindex,nofollow' to prevent search engine indexing of ListMail pages
 - added overwrite active and custom refurl options to user import process
 - added referring url to user selection
 - added 'number of subscribers' message code with optional offset value
 - added user status, ip address, referring url and html fields plus new web export options to export process
 - added confirmation reminder emails - can be set to send every X days to a maximum of X messages
 - fixed multi-list remove via form - now correctly removes users even if already removed from one or more selected lists
 - improved login. if you try to access a ListMail page when not logged in you are now brought to the login page
  + the url you tried to access is saved and automatically redirected to upon login. (bookmark your internal urls!)
 - added warning if more than 50% of messages are 'skipped' with SMTP due to auth errors not being detected on some servers
 - rewrote backup & restore process - linux mysql command-line utilities are no longer required, backup should work on Windows
  + current list is no longer lost when writing backup files
 - added new 'Upload' option to backup process to promote new backup service.
  + set $no_auto_backup=1; in config.php to disable
 - fixed PHP uploads, must use $_FILES instead of $HTTP_POST_FILES for later versions of PHP
 - fixed mysql connection failure on servers with slow DNS response for invalid domains during SMTP
  + default socket timeout is 9s, can set $smtp_timeout var in config.php to override
 - added ability to enable SMTP debug log globally
 - added ability to make custom fields required, complete with user-defined error messages
 - added ability to specify custom seq and delay with loop followups
 - improved multi-list subscribe - success if dupe on one list but not on at least one other
  + error if missing required field on any list
  + list 'additional options' targeting lists being subscribed to ignored
 - fixed a rare issue with GET-style vars on certain servers. if last var was empty no vars were posted
 - fixed inability to rename error message sets under FireFox

-----------------------------------------------------------------

Changes v1.86 -> v1.86b

 - fixed dailymail 'skip bounce via mailbox' option - mailbox processing was not running in v1.85+
 - fixed saving of certain search rules in user selection

-----------------------------------------------------------------

Changes v1.85 -> v1.86

 - added ability to add, name, save, load, and edit entire sets of followups
 - added ability to schedule application of followup sets
 - added ability to send more than one followup on the same day with duplicate seq #s
 - added 'loop followups' list option to automatically restart users at end of sequence to first followup
 - added 'reset+load saved followup set when x% users are at end of sequence' list option
 - added list options to remove/add users from/to other lists when removed/added to list
 - added offsets to date-based message codes, can specify +/- days, months, years, and combinations
 - added 'set to default' option to custom html, default html resets to system default others to default html
 - added Message-Id: header to SMTP emails in accordance with internet email standards
 - removed zip compression from backup and restore due to inability to overcome memory limits
 - added gzip compression to backup AND restore with memory buffer for very large files
 - added delay # to user selection searches
 - improved 'reset users' to allow setting all users on current list to a chosen followup and delay
 - improved list settings page when deleting lists, now sets current list to next lower or first list
 - fixed users who subscribe, do not confirm, and then resubscribe again now keep the same uid
 - fixed users who click confirm twice are now shown a 'duplicate email' error and do not receive another email
 - fixed occasional FireFox display width issue on message editing page when Date showing
 - fixed quotes in filenames causing inability to apply attachments
 - fixed file upload issue on some restrictive web hosts - imports are now temp copied to ./attach
 - fixed signup-xtra.php optional custom script processing during import
 - fixed dailymail report, now sent when a mailing is completed even if smtp error or server timeout
 - fixed sent message maintenance, messages are now properly removed
 - removed aol.com from text-only default
 - added ability to disable new dailymail 'once every 23h55m' protection

-----------------------------------------------------------------


config.php
<?php
// CONFIG.PHP - ListMail Configuration File
//
// This is a .PHP file that must maintain proper syntax.  Be careful to only
// change the values between the single quotes and nothing else, or the script
// may not work correctly.
//
// Required is your database information, the host should normally be left at the
// setting, 'localhost', but not always.  Check with your host if it doesn't work.

// Enter your MySQL Information!
$sqlhost = 'localhost';
$sqluser = 'listmail';
$sqlpass = 'listmail';
$sqldb = 'listmail';

// SAVE this file as config.php, -not- config.php.txt
// UPLOAD all of the files to a new folder on your web site, such as public_html/mail/
// BROWSE to the new subdirectory

// OPTIONS are available for different server types
// These options are rarely needed but are available if problems occur
// To change options remove and add the comment characters "//" from the beginning of the lines.
// Examples are provided with their defaults
// Setting an option to '' means no value, usually off.  Be careful to keep proper PHP syntax.

// Detect line endings for Mac browsers, default 0
// ini_set('auto_detect_line_endings',1);

// Use -f setting to set return-path correctly with PHP mail()
// This is used for welcome and confirmation messages.  If you are receiving "raw" bounce
// messages from welcome emails even if you have bounce configured this may need to be set.
 $phpmailf = '';
// $phpmailf = 1;

// International character set, default iso-8859-1 if not set)
$charset = '';
// $charset = 'iso-8859-1';

// Manually set path to MySQL binaries - for problems with backup / restore.
// The linux commandline utilities "mysql" and "mysqldump" need to be available for the
// backup process.  If ListMail cannot detect the path you can set it manually.

// $sqlpath = "/usr/bin/mysql/";
// $sqlpath = "/usr/local/bin/mysql/";
// $sqlpath = "/usr/local/mysql/bin/";
// $sqlpath = "/usr/local/psa/mysql/bin/";

// max random value for problematic servers (deprecated)
// $randmaxx = '50000';

// mail counter increment, default 50
// $mailcount = 50;

// path for mimeclass/mimepart (sendwelcome)
// blank is normal - use current directory always '.' comment to allow setting $mp from your script
$mp = '';

// add Maximizer option to exports
/* $maximizer = 1;
$max_header = '"MTI: Web Inquiry"
"BUSINESS_PEOPLE","1","401","E-mail Address"
"P_E-mail Address","FIRST NAME","LAST NAME","ADDRESS LINE 1","ADDRESS LINE 2","CITY","COUNTY","POSTCODE","COUNTRY","PHONE 1"
"ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-39","ALPHA-59","ALPHA-21","ALPHA-21","ALPHA-21","ALPHA-119"';
$max_fields = 'em,fn,ln,u1,u2,u3,u4,u5,u6,u7'; */

// !! Do not change anything below here !!
if (get_magic_quotes_gpc()==1) { $gpc = 'yes'; }
?>
